README.md

```diff
@@ -9,7 +9,7 @@
 
 ### Dependencies
 
-- JDK 17 (ex: openjdk-17-jdk on Ubuntu)
+- JDK 21 (ex: openjdk-21-jdk on Ubuntu)
```

Comment:
This doesn't actually make a difference.

Gradle itself is gonna run on your JDK of preference. It should run on JDK 17 or 21 fine.
However we are facing some issues with verify DB migrations on JDK 21.

With the change to use `jvmToolchain` settings, Gradle will download the necessary JDK when compiling/running Java tasks.